src/Common/src/System/Net/DebugCriticalHandleMinusOneIsInvalid.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System.Net
+{
+#if DEBUG
+    //
+    // This is a helper class for debugging GC-ed handles that we define.
+    // As a general rule normal code path should always destroy handles explicitly
+    //
+    internal abstract class DebugCriticalHandleMinusOneIsInvalid : CriticalHandleMinusOneIsInvalid
+    {
```

Comment:
The only reason is that most of these classes do logging that is specific to System.Net and so can't be shared unless the code takes dependency on our GlobalLog ETW tracer.